src/main/java/fi/aalto/cs/apluscourses/model/ModelFactory.java

```diff
@@ -0,0 +1,14 @@
+package fi.aalto.cs.apluscourses.model;
+
+import java.net.URL;
+import java.util.List;
+import java.util.Map;
+import org.jetbrains.annotations.NotNull;
+
+public interface ModelFactory {
+  Course createCourse(@NotNull String name,
+                      @NotNull List<Module> modules,
+                      @NotNull Map<String, String> requiredPlugins);
+
+  Module createModule(@NotNull String name, @NotNull URL url);
```

Comment:
Nit (can be fixed later in some other PR): return values could have `NotNull` as well